scripts/common/vibe-governance-helpers.ps1

```
@@ -670,37 +670,122 @@ function Resolve-VgoPathSpec {
+
+    return [pscustomobject]@{
+        preferred_powershell_host = if ($rawPolicy.PSObject.Properties.Name -contains 'preferred_powershell_host' -and -not [string]::IsNullOrWhiteSpace([string]$rawPolicy.preferred_powershell_host)) { [string]$rawPolicy.preferred_powershell_host } else { [string]$defaults.preferred_powershell_host }
+        require_pwsh_on_non_windows = if ($rawPolicy.PSObject.Properties.Name -contains 'require_pwsh_on_non_windows') { [bool]$rawPolicy.require_pwsh_on_non_windows } else { [bool]$defaults.require_pwsh_on_non_windows }
+        allow_windows_powershell_fallback = if ($rawPolicy.PSObject.Properties.Name -contains 'allow_windows_powershell_fallback') { [bool]$rawPolicy.allow_windows_powershell_fallback } else { [bool]$defaults.allow_windows_powershell_fallback }
+        record_host_resolution_artifacts = if ($rawPolicy.PSObject.Properties.Name -contains 'record_host_resolution_artifacts') { [bool]$rawPolicy.record_host_resolution_artifacts } else { [bool]$defaults.record_host_resolution_artifacts }
+    }
+}
+
+function Resolve-VgoPowerShellHost {
+    $policy = Get-VgoPowerShellHostPolicy
+    $isWindows = [System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform([System.Runtime.InteropServices.OSPlatform]::Windows)
```

PSScriptAnalyzer (1.25.0)

[error] Starting from PowerShell 6.0, the Variable 'isWindows' cannot be assigned any more since it is a readonly automatic variable that is built into PowerShell, please use a different name.

(PSAvoidAssignmentToAutomaticVariable)